app/controllers/holds_controller.rb

```diff
@@ -18,14 +18,15 @@ def index
   #
   # PATCH /holds
   # PUT /holds
-  def update
-    params['hold_list'].each do |holdkey|
-      @hold_to_act_on = holds.find { |hold| hold.key == holdkey }
-      handle_pickup_change_request if params['pickup_library'].present? && params['pickup_library'].present?
-      handle_not_needed_after_request if params['pickup_by_date'].present?
+  def batch_update
+    params['hold_list'].each do |hold_key|
+      ws_args = { hold_key: hold_key,
+                  pickup_library: params[:pickup_library],
+                  pickup_by_date: params[:pickup_by_date],
+                  session_token: current_user.session_token }
+      ChangePickupLibraryJob.perform_later(ws_args) if params[:pickup_library].present?
+      handle_not_needed_after_request(ws_args) if params[:pickup_by_date].present?
     end
-
-    redirect_to holds_path
```

Comment:
Rails responds with a "204" no changes. This OK, I think. There are some other options like `render head: ok` or something like that